working_dir.go

```diff
@@ -116,7 +116,6 @@ func (wd *WorkingDir) RequireClearPlan(t TestControl) {
 func (wd *WorkingDir) init(pluginDir string) error {
 	args := []string{"init"}
 	args = append(args, wd.baseArgs...)
-	args = append(args, "-plugin-dir="+pluginDir, wd.configDir)
```

Comment:
Was this removal intentional? 👀

Reply:
Yes, this removal is the key change in this PR, allowing Terraform CLI to use the normal plugin discovery mechanism. Currently, if `-plugin-dir` is set, then Terraform CLI will search _only_ that directory for plugins.

An earlier solution involved changing this behaviour when the `-get-plugins` flag is explicitly set to `true`. Please see https://github.com/hashicorp/terraform/pull/23982 for a discussion of how these flags work.

If we're happy with this removal then the Core PR can be closed, as it is not a trivial change in functionality.